flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowJoinUtil.scala

```diff
@@ -127,7 +127,7 @@ object WindowJoinUtil {
       case None =>
         s"""
            |$buildJoinedRow
-           |$collectorTerm.collect($joinedRow)
+           |$collectorTerm.collect($joinedRow);
```

Comment:
I looked into the code and find that we will never enter this branch under the current implementation, because `otherCondition` is never null (always contains join key). That means we can't reproduce this problem currenlty. If we want to reproduce this probem, we have to refactor the window join (esp. `otherCondition`). However, this is a major work.

My suggestion would be: simply add the semicolon and do not add tests, and refactor window join in following issues, I just create FLINK-15441 to track this.

Reply:
Thank you @wuchong, I know what you mean, I have deleted unnecessary test.